opennms-doc/guide-admin/src/asciidoc/text/minion/kafka.adoc

```diff
@@ -1,112 +0,0 @@
-
-The _Kafka_ server must be compatible with _Kafka_ client version `1.0.1`.
-
-==== Consumer Configuration
-
-Enable and configure the _Kafka_ consumer on _{opennms-product-name}_ by using the following commands. The `initialSleepTime` property will ensure that messages are not consumed from _Kafka_ until the _{opennms-product-name}_ system has fully initialized.
-
-[source, sh]
-----
-echo 'org.opennms.core.ipc.sink.initialSleepTime=60000' > "$OPENNMS_HOME/etc/opennms.properties.d/sink-initial-sleep-time.properties"
-echo 'org.opennms.core.ipc.sink.strategy=kafka
-org.opennms.core.ipc.sink.kafka.bootstrap.servers=127.0.0.1:9092' >> "$OPENNMS_HOME/etc/opennms.properties.d/kafka.properties"
-----
-
-Restart _{opennms-product-name}_ to apply the changes.
-
-Additional _Kafka_ consumer options can be set by defining additional system properties prefixed with `org.opennms.core.ipc.sink.kafka`.
-For example, you can customize the group ID using `org.opennms.core.ipc.sink.kafka.group.id=MyOpenNMS`.
-
-A list of all the available options can be found here in link:https://kafka.apache.org/10/documentation.html#newconsumerconfigs[New Consumer Configs].
-
-[[ga-minion-kafka-producer-configuration]]
-==== Producer Configuration
-
-Enable the _Kafka_ producer on _Minion_ using:
-
-[source, sh]
-----
-echo '!opennms-core-ipc-sink-camel
-opennms-core-ipc-sink-kafka' >> "$MINION_HOME/etc/featuresBoot.d/kafka.boot"
-----
-
-NOTE: The snippet above prevents the `opennms-core-ipc-sink-camel` feature from starting when _Minion_ is started, and loads the `opennms-core-ipc-sink-kafka` feature instead.
-
-Next, configure the _Kafka_ producer on _Minion_ using:
-
-[source, sh]
-----
-echo 'bootstrap.servers=127.0.0.1:9092
-acks=1' > "$MINION_HOME/etc/org.opennms.core.ipc.sink.kafka.cfg"
-----
-
-Restart _Minion_ to apply the changes.
-
-Additional _Kafka_ producer options can be set directly in the `org.opennms.core.ipc.sink.kafka.cfg` file reference above.
-A list of all the available options can be found here in link:https://kafka.apache.org/10/documentation.html#producerconfigs[Producer Configs].
-
-=== Using Kafka for RPC
-
-By default, _{opennms-product-name}_ uses the embedded _ActiveMQ_ broker to communicate with _Minions_.
-Enabling kafka for RPC will allow replacing _ActiveMQ_ if needed.
-
-_Kafka_ must be enabled on both _{opennms-product-name}_ and _Minion_ to function.
-
-The _Kafka_ server must be compatible with _Kafka_ client version `1.0.1`.
-
-NOTE: For Kafka RPC, number of partitions should always be greater than number of minions at a location. When there are multiple locations, partitions >= max (number of minions at a location).
-
-NOTE: By default, kafka RPC supports buffers greater than >1MB by splitting large buffer into chunks of 900KB(912600). Max buffer size (900KB, by default) can be configured by setting org.opennms.core.ipc.rpc.kafka.max.buffer.size ( in bytes).
```

Comment:
That's true, but besides using the producer level setting `max.buffer.size`, each Kafka broker requires the following broker level settings: `message.max.bytes`, `replica.fetch.max.bytes`, and I also recommend `compression.type=producer`.

This is crucial for the Kafka Producer Feature, but not necessarily for Sink and RPC as newer versions support split messages into smaller chunks to avoid the message size limits, but I prefer to have those settings present.

Reply:
Can you help me here, you have added the comment for text which is removed.

Reply:
For the reason they are not required to get you going with a Kafka setup and to avoid to clutter the step by step instructions with a lot of hints here, notes there, I've collected them in this section here. The focus in this section is the RPC and Sink feature with Kafka. I would say, the integration with Kafka for integration to have OpenNMS a producer for Alarms, Metrics and Inventory data should be a different topic in my opinion. What do you think=

Reply:
I'll assume I haven't explained myself correctly, so I'll try again.

Forget that I've mentioned the Kafka Producer.

Your work mentions the Max buffer size limitation. If that is going to be mentioned for RPC/Sink, then it must be explained correctly, meaning the mandatory broker-level settings should be there (as I mentioned).

That said, on newer versions of Horizon, RPC and Sink support split messages into chunks, so in theory, there is no need to mention the message size limits, although, it is always a good practice to keep those settings in mind and set some defensive values to avoid issues.

Makes sense?

Reply:
Really sorry, but I can't really follow what you want to have to be changed. The section which has this conversation here is removed. To avoid clutter and not losing the information, I've moved it into a [Tuning Apache Kafka](https://github.com/OpenNMS/opennms/pull/2797/files#diff-22310b44ed7aca0958e6a179d1968f2bR2) section which can be extended and maintained separately.

If there is something wrong or need to be improved here, feel free and just edit and commit directly to this branch so we can merge it.